client/src/components/feedback/edit-question-for-queue-modal.tsx

```diff
@@ -0,0 +1,271 @@
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import {
+  Backdrop,
+  Button,
+  Fade,
+  FormControl,
+  Grid,
+  InputLabel,
+  makeStyles,
+  Modal,
+  Select,
+  TextField,
+  Theme,
+  Typography,
+} from "@material-ui/core";
+import React, { useState } from "react";
+import {
+  Category,
+  Mentor,
+  Question,
+  QuestionType,
+  Subject,
+  Topic,
+  UtteranceName,
+} from "types";
+import { onTextInputChanged } from "helpers";
+import { v4 as uuid } from "uuid";
+import { Autocomplete } from "@material-ui/lab";
+import { addOrUpdateSubjectQuestions, addQuestionToRecordQueue } from "api";
+import { SubjectQuestionGQL } from "types-gql";
+
+const useStyles = makeStyles((theme: Theme) => ({
+  homeThumbnail: {
+    position: "relative",
+    width: "78%",
+    height: 180,
+  },
+  siteThumbnail: {
+    width: 180,
+    height: 135,
+  },
+  title: {
+    fontWeight: "bold",
+  },
+  inputField: {
+    width: "100%",
+    margin: 10,
+  },
+  modal: {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  paper: {
+    backgroundColor: theme.palette.background.paper,
+    border: "2px solid #000",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+    maxWidth: "50%",
+  },
+}));
+
+function EditQuestionForQueueModal(props: {
+  handleClose?: () => void;
+  open: boolean;
+  mentor: Mentor;
+  userQuestion?: string;
+  accessToken: string;
+}): JSX.Element {
+  const { handleClose, open, userQuestion, mentor, accessToken } = props;
+  const classes = useStyles();
+  const [selectedSubject, setSelectedSubject] = useState<Subject>();
+  const [selectedCategory, setSelectedCategory] = useState<Category>();
+  const [selectedTopic, setSelectedTopic] = useState<Topic>();
+  const [customQuestion, setCustomQuestion] = useState<string>(
+    userQuestion || ""
+  );
+
+  function okButtonClicked(customQuestion: string) {
+    // create new question
+    const newQuestion: Question = {
+      _id: uuid(),
+      question: customQuestion,
+      type: QuestionType.QUESTION,
+      name: UtteranceName.NONE,
+      clientId: uuid(),
+      paraphrases: [],
+      mentor: mentor._id,
+    };
+    // create newSubjectQuestion : to add to db
+    const emptyTopic: Topic = { id: "", name: "", description: "" };
```

Comment:
This won't work and shouldn't be done, this will throw errors since empty id's are not allowed and we shouldn't be creating "empty" documents

Reply:
A topic list is required to create a Subject Question, but topics are optional fields in the modal. How do we handle that? Either we make topics required or use a different method in adding the new question to the db?